src/app/features/psbt-signer/hooks/use-sign-psbt-error.tsx

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react';
+
+import { RpcErrorCode } from '@btckit/types';
+
+import { finalizePsbt } from '@shared/actions/finalize-psbt';
+import { makeRpcErrorResponse } from '@shared/rpc/rpc-methods';
+
+import {
+  usePsbtRequestSearchParams,
+  useRpcSignPsbtParams,
+} from '@app/common/psbt/use-psbt-request-params';
+
+export function useSignPsbtError() {
+  const { requestToken, tabId } = usePsbtRequestSearchParams();
+  const { requestId, tabId: rpcTabId } = useRpcSignPsbtParams();
+
+  return useCallback(
+    (errorMsg: string) => {
+      if (requestToken)
+        finalizePsbt({
+          requestPayload: requestToken,
+          tabId,
+          data: errorMsg,
+        });
+      chrome.tabs.sendMessage(
+        rpcTabId,
+        makeRpcErrorResponse('signPsbt', {
+          id: requestId,
+          error: {
+            message: errorMsg,
+            code: RpcErrorCode.INTERNAL_ERROR,
+          },
+        })
+      );
+      window.close();
```

Comment:
should we keep the window open when there is an error?

Reply:
Is that preferred, and show an error msg to the user?

Reply:
imo yes - this is also similar to how Xverse wallet handles errors.

Reply:
@mica000 based on this feedback, we will need a way to show the caught errors in the UI for the user. It can just be at the bottom though like our other error placement for app transactions.